modules/images/image-loading-optimization/detection/detect.js

```diff
@@ -175,20 +165,55 @@ export default async function detect( {
 		return;
 	}
 
-	// Prevent detection when page is not scrolled to the initial viewport.
-	// TODO: Does this cause layout/reflow? https://gist.github.com/paulirish/5d52fb081b3570c81e3a
-	if ( doc.documentElement.scrollTop > 0 ) {
+	// Abort if the current viewport is not among those which need URL metrics.
+	if ( ! isViewportNeeded( win.innerWidth, neededMinimumViewportWidths ) ) {
 		if ( isDebug ) {
-			warn(
-				'Aborted detection since initial scroll position of page is not at the top.'
-			);
+			log( 'No need for URL metrics from the current viewport.' );
 		}
 		return;
 	}
 
-	if ( ! isViewportNeeded( win.innerWidth, neededMinimumViewportWidths ) ) {
+	// Ensure the DOM is loaded (although it surely already is since we're executing in a module).
+	await new Promise( ( resolve ) => {
+		if ( doc.readyState !== 'loading' ) {
+			resolve();
+		} else {
+			doc.addEventListener( 'DOMContentLoaded', resolve, { once: true } );
+		}
+	} );
+
+	// Wait until the resources on the page have fully loaded.
+	await new Promise( ( resolve ) => {
+		if ( doc.readyState === 'complete' ) {
+			resolve();
+		} else {
+			win.addEventListener( 'load', resolve, { once: true } );
+		}
+	} );
+
+	// Wait yet further until idle.
+	if ( typeof requestIdleCallback === 'function' ) {
+		await new Promise( ( resolve ) => {
+			requestIdleCallback( resolve );
+		} );
+	}
+
+	// As an alternative to this, the ilo_print_detection_script() function can short-circuit if the
+	// ilo_is_url_metric_storage_locked() function returns true. However, the downside with that is page caching could
+	// result in metrics being missed being gathered when a user navigates around a site and primes the page cache.
```

Comment:
"result in metrics being missed being gathered" is confusing

Reply:
```suggestion
	// result in metrics missed from being gathered when a user navigates around a site and primes the page cache.
```

How's this?

Reply:
much better, thanks